src/lib/logical_query_plan/stored_table_node.cpp

```diff
@@ -89,6 +90,56 @@ bool StoredTableNode::is_column_nullable(const ColumnID column_id) const {
   return table->column_is_nullable(column_id);
 }
 
+std::vector<IndexStatistics> StoredTableNode::indexes_statistics() const {
+  DebugAssert(!left_input() && !right_input(), "StoredTableNode must be a leaf");
+
+  const auto table = StorageManager::get().get_table(table_name);
+  auto stored_indexes_statistics = table->indexes_statistics();
+
+  if (_pruned_column_ids.empty()) {
+    return stored_indexes_statistics;
+  }
+
+  auto column_pruned_bitvector = std::vector<bool>();
+  auto column_left_shifts = std::vector<ColumnID>();
+
+  column_pruned_bitvector.resize(table->column_count());
+  column_left_shifts.resize(table->column_count());
+
+  // Fill the bitvector
+  for (const auto& pruned_column_id : _pruned_column_ids) {
+    column_pruned_bitvector[pruned_column_id] = true;
+  }
+
+  // Calculate left_shifts
```

Comment:
(Not a full PR, just trying to get an overview of the open PRs)

This whole "adapt a vector to remove the pruned items" seems like something that is not needed only here. Can we make it more abstract?

Reply:
It's rather "build a mapping structure which gives me the ColumnID of the pruned table as output for a provided original ColumnID as input" than "adapt a vector to remove the pruned items". I haven't found any similar easy to use procedure in the current code base yet. However, I agree with you that this procedure should be made reusable.